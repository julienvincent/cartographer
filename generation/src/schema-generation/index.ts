export * as litematica from './litematica';
