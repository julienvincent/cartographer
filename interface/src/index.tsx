import * as ReactDOM from 'react-dom';
import * as React from 'react';
import * as Root from './root';

// import 'nes.css/css/nes.min.css';
import './index.css';

ReactDOM.render(
  <React.StrictMode>
    <Root.Component />
  </React.StrictMode>,
  document.getElementById('root')
);
