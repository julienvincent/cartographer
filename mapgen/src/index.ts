export * as schema_generation from './schema-generation';
export * as block_generation from './block-generation';
export * as serialization from './serialization';
