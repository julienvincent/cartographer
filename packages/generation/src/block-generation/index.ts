export * from './block-space';
export * from './staircase';
