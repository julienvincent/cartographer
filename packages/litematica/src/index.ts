export * as litematica_bit_array from './litematica-bit-array';
export * as long from './long';
export * from './generator';
