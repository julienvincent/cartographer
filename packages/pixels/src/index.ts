export * as transformers from './transformers';
export * as conversion from './conversion';
export * from './definitions';
