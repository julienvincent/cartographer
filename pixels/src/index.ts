export * as conversion from './conversion';
export * as defs from './defs';
