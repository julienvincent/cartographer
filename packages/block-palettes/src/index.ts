import { BlockPalette } from '@cartographer/pixels';

import palette_1_20 from './palettes/1.20.json';
import palette_1_19 from './palettes/1.19.json';
import palette_1_18 from './palettes/1.18.json';

export const palettes = {
  '1.20': palette_1_20 as unknown as BlockPalette,
  '1.19': palette_1_19 as unknown as BlockPalette,
  '1.18': palette_1_18 as unknown as BlockPalette
};
