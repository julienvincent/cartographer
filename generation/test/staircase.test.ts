import * as block_generation from '../src/block-generation';
import 'jest';
import { test, expect, describe } from '@jest/globals';

describe('optimiseStaircase', () => {
  test('it should work', () => {
    expect(block_generation.optimiseStaircase([2, 1, 2, 3, 2, 1, 0])).toEqual([1, 0, 1, 3, 2, 1, 0]);
    expect(block_generation.optimiseStaircase([2, 2, 3, 4, 6, 6, 6, 6, 6, 7])).toEqual([0, 0, 1, 2, 3, 3, 3, 3, 3, 4]);
  });
});
