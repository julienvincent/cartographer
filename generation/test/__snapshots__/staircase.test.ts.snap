// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`optimiseStaircase it should correctly scale a PixelGrid down 1`] = `
Array [
  1,
  2,
]
`;

exports[`pixel conversions it should correctly scale a PixelGrid down 1`] = `
Array [
  1,
  2,
]
`;
